src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs

```diff
@@ -1214,10 +1251,12 @@ private async Task AttachToTarget(SessionId sessionId, CancellationToken token)
             // see https://github.com/mono/mono/issues/19549 for background
             if (sessions.Add(sessionId))
             {
+                string checkUncaughtExceptions = "throw \"pause_on_uncaught\";";
+                string checkCaughtExceptions = "try {throw \"pause_on_caught\";} catch {}";
                 await SendMonoCommand(sessionId, new MonoCommands("globalThis.dotnetDebugger = true"), token);
                 Result res = await SendCommand(sessionId,
                     "Page.addScriptToEvaluateOnNewDocument",
-                    JObject.FromObject(new { source = "globalThis.dotnetDebugger = true; delete navigator.constructor.prototype.webdriver" }),
+                    JObject.FromObject(new { source = $"globalThis.dotnetDebugger = true; delete navigator.constructor.prototype.webdriver; {checkUncaughtExceptions} {checkCaughtExceptions}" }),
```

Comment:
I'm not sure I understand the flow here.

1. on `attach`, we throw two exceptions, so we can infer the pause-on-exceptions setting of the debugger;
2. then I see two code paths:
  a. `Runtime.exceptionThrown` - where we do the inference, and set the value accordingly, in the proxy, and the app, and only when runtime is not ready
  b. `Debugger.paused` - where we resume, if it paused because of one of the above exceptions

- so, what are the actual scenarios here?
- And what does the flow of events look like, for them?

Also, it would be very useful to express that in tests too, but that might not be straight forward, so it can be in a follow up PR.

Reply:
If we attach to an already loaded page it will only pass on Runtime.exceptionThrown and this would print a message on console if I don't return true.
If we refresh a page with the debugger attached it will pass on Debugger.paused, and we run the resume, and it will also pass on Runtime.exceptionThrown that would print a message if I don't return true.
In both cases before send the runtime ready we have already got the configuration so we can set it to mono debugger.

Reply:
Oh no, I tested again, when the page is already loaded and we attach the debugger we receive the "Debugger.setPauseOnExceptions" with the correct information.
When we refresh the page that we are debugging, then we will receive the Debugger.paused and the Runtime.exceptionThrown.

Reply:
> If we attach to an already loaded page it will only pass on Runtime.exceptionThrown and this would print a message on console if I don't return true.

By "pass on foo", do you mean that `foo`'s code path will be followed?

Question for scenario #1:

1. attach to already loaded page
2. we send a `Page.addScriptToEvaluateOnNewDocument`, with `.. {checkUncaughtExceptions} {checkCaughtExceptions}`
3. and based on your last comment, we get `Debugger.setPauseOnExceptions`
4. And then .. do we get `Runtime.exceptionThrown`? If yes, then for which case(caught/uncaught)?

> If we refresh a page with the debugger attached it will pass on Debugger.paused, and we run the resume, and it will also pass on Runtime.exceptionThrown that would print a message if I don't return true.

- Also, what happens for the `{checkUncaughtExceptions}` in the command that we send.
  - if the setting is `uncaught`, then debugger is paused, we get `exceptionThrown, resume the debugger. Does the code then resume to the next line, and we execute `checkCaughtExceptions`?

- Is there any case where we might want/need to get the "current" setting from the app, and use that in the proxy?

Reply:
I changed somethings in the PR.
For scenario #1:
We get `Runtime.exceptionThrown` in the case of the page already loaded for uncaught and we don't receive `Debugger.Pause`. Then we receive the `Debugger.setPauseOnExceptions` with the correct information.

For scenario #2:
If the setting is `uncaught` and we refresh the page we get:
Debugger.pause where we set the context status that will be send later to mono runtime. And the we receive a `Runtime.exceptionThrown` that will just return true to avoid printing the exception on the console.
If the setting is `all` and we refresh the page we get:
Debugger.pause where we set the context status PauseOnCaught to true then we resume, then we receive another Debugger.pause where we set the context status PauseOnUncaught to true then we resume. Then we receive `Runtime.exceptionThrown` for uncaught.
If while loading the page before the runtime is ready we change the status, we will receive a `Debugger.setPauseOnExceptions` with the right information, so I override the information that is in context for PauseOnCaught and PauseOnUncaught and when the runtime is ready I send the correct status.

We don't need the current setting from the app, the current setting should always be correct, because during the use of the debugger we receive the `Debugger.setPauseOnExceptions` with the correct information.